pyvkfft/test/test_fft.py

```diff
@@ -412,6 +413,44 @@ def test_pycuda_streams(self):
                     dn = np.roll(d, i * 7, axis=1)
                     self.assertTrue(np.allclose(dn, vd[i].get(), rtol=rtol, atol=abs(dn).max() * rtol))
 
+    @unittest.skipIf(not has_pyopencl, "pyopencl is not available")
+    def test_pyopencl_queues(self):
+        """
+        Test multiple FFT with queues different from the queue used
+        in creating the VkFFTApp
+        """
+        for dtype in (np.complex64, np.complex128):
+            with self.subTest(dtype=np.dtype(dtype)):
+                init_ctx("pyopencl", gpu_name=self.gpu, verbose=False)
+                ctx = gpu_ctx_dic["pyopencl"][2].context
+                if dtype == np.complex64:
+                    rtol = 1e-6
+                else:
+                    rtol = 1e-12
+                sh = (256, 256)
+                d = (np.random.uniform(-0.5, 0.5, sh) + 1j * np.random.uniform(-0.5, 0.5, sh)).astype(dtype)
+                n_queues = 5
+                vd = []
+                vapp = []
+
+                with cl.CommandQueue(ctx) as cq:
```

Comment:
Creating the queue using `with` may be dangerous here. This means the queue should not be used anymore outside of the `with`block, but a reference to that queue will still persist in the `VkFFTApp` object. It is not used later in the code so it is fine here, but I think it's giving a dangerous example.

Reply:
Yes, I agree that it's dangerous to keep persisting the queue in `VkFFTApp`. Latest pyopencl will give a warning if the queue is used outside of the with block though. https://github.com/inducer/pyopencl/pull/561